### PR TITLE
Add splitPathComponents and tests

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Create tests/stdlib/tospaths.nim file

### CODE REVIEW

**Comment by Araq** [comment]
The test is not run. "testament" tests should not use "unittest". I don't see the need for splitPathComponents, do we really need it?

**Comment by GULPF** [comment]
@Araq Imo it would be nice if it was possible to use `unittest` with the tester in a sane way. Writing tests for complex stdlib modules with `doAssert` gets old fast.

**Comment by FedericoCeratto** [comment]
Walking forth and back across a path using splitPath is cumbersome and requires multiple calls.
Confusingly, there are around 10 files named tests/stdlib/t*.nim that use unittest. There is no description on where and how to enable the tests in readme.md

**Comment by Araq** [comment]
@GULPF unittest is not a sane way to do testing IMO, regardless of how well it would "work" with testament.

File: tests/stdlib/tospaths.nim [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
+    doAssert unixToNativePath("../../abc") == "../../abc"
+    doAssert unixToNativePath("/abc", "a") == "/abc"
+    doAssert unixToNativePath("/abc/def", "a") == "/abc/def"
+
+doAssert splitPathComponents("").len == 0
+doAssert splitPathComponents("/").len == 0
+doAssert splitPathComponents("foo") == @["foo"]
+doAssert splitPathComponents("/foo") == @["foo"]
+doAssert splitPathComponents("foo/bar") == @["foo", "bar"]
+doAssert splitPathComponents("/foo/bar") == @["foo", "bar"]
```
Comment (dom96) [inline tests/stdlib/tospaths.nim]: Isn't ``/`` a component? It's the root directory. For Windows ``C:\foo\bar`` should give ``["C:\", "foo", "bar"]`` no?

Comment (citycide) [reply tests/stdlib/tospaths.nim]: Having tests for Windows paths would be appreciated now that you mention this.

**Comment by Araq** [comment]
I've a better implementation ready in the compiler and will move it to ``ospaths``.

**Pull request closed without merging**